tests/framework/base/ComponentTest.php

```diff
@@ -445,6 +445,25 @@ public function testDetachNotAttachedHandler()
         $this->assertFalse($obj->off('test', [$this, 'handler2']), 'Trying to remove the handler that is not attached');
         $this->assertTrue($obj->off('test', [$this, 'handler']), 'Trying to remove the attached handler');
     }
+
+    /**
+     * @see https://github.com/yiisoft/yii2/issues/17223
+     */
+    public function testEventClosureDetachesItself()
+    {
+        if (PHP_VERSION_ID < 70000) {
+            $this->markTestSkipped('Can not be tested on PHP < 7.0');
+            return;
+        }
+
+        $obj = include 'stub/AnonymousComponentClass.php';
```

Comment:
```suggestion
        $obj = require __DIR__ . '/stub/AnonymousComponentClass.php';
```

Reply:
Should we improve this code too?
https://github.com/yiisoft/yii2/blob/caf9a28657eca8e49ead3d79a9b24a2439f5a829/tests/framework/base/ModelTest.php#L497

Reply:
Yes.